sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosSchedulers.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+public class CosmosSchedulers {
+    private final static String COSMOS_PARALLEL_THREAD_NAME =  "cosmos-parallel";
+
+    // Using a custom parallel scheduler to be able to schedule retries etc.
+    // without being vulnerable to scenarios where applications abuse the
+    // Parallel scheduler and cause thread starvation on Reactor Core parallel scheduler
+    public final static Scheduler Parallel = Schedulers.newParallel(
```

Comment:
does this need to get released?

are the underlying threads daemon threads?

 does the presence of this prevent the java app from terminating if it is not released?

Reply:
@moderakh  - at least what I got from Reactor code is that this will behave exactly as `Schedulers.parallel()` was working before. 
See this code from reactor-core `Schedulers` library. 
```
static final Supplier<Scheduler> PARALLEL_SUPPLIER = () -> {
        return newParallel("parallel", DEFAULT_POOL_SIZE, true);
    };
```
@FabianMeiswinkel  - please correct me if I am wrong.

Reply:
Like for Reactor core's parallel thread pool it uses daemon threads (last parameter) - so it won't terminating the app and I don't think there is a need to release it.